HOTFIX: fix broken Streams upgrade system test

mjsax opened the pull request:
Fixes system testing bug. Regression introduced via https://github.com/apache/kafka/pull/12122

This PR must be cherry-picked back to 3.3

mjsax committed:
HOTFIX: fix broken Streams upgrade system test

vcrfxia reviewed (approved):
Nice catch, @mjsax !

mjsax commented:
Triggered a system test run: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/5653/

mjsax commented:
> Nice catch

Well, rather embarrassing... I reviewed and merge the PR introducing the issue...

mjsax commented:
\cc @mimaison as RM for 3.5.0 -- I hope we did not introduce any issues without noticing it... 🤞 -- but if we find anything, it could block/delay 3.5.0...

mjsax commented:
Timeout:
```
16:34:05 An error occurred while executing multiple actions in parallel.
16:34:05 Any errors that occurred are shown below.
16:34:05 
16:34:05 An error occurred while executing the action on the 'worker27'
16:34:05 machine. Please handle this error then try again:
16:34:05 
16:34:05 The instance never became "ready" in AWS. The timeout currently
16:34:05 set waiting for the instance to become ready is 120 seconds.
```

Re-triggered the run.

mjsax commented:
Same issue. Trying again: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/5655/

mjsax committed:
fix dev version

mjsax commented:
System test failed due to incorrect dev-version. Pushed a fix.

Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/5656/

mjsax commented:
13 tests failed -- `StreamsBrokerBounceTest` should be unrelated, but also `StreamsUpgradeTest.test_rolling_upgrade_with_2_bounces`

Re-triggred to see if the same tests fails, or if it's flakyness: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/5657/

mjsax committed:
fix test setup

mjsax commented:
Ok. Dug into this a little bit, and some test are broken, because KS is broken.

The original PR for https://issues.apache.org/jira/browse/KAFKA-13769 broke KS, but it also added the test regression using randomized `application.id`s and thus masked the error it introduced. The issue was reported later via https://issues.apache.org/jira/browse/KAFKA-14646 and we fixed it forward.

However, the fact remains that rolling bounce upgrades are broken for 2.4 - 3.0 release as "from_version". I updates this PR to disable the corresponding tests.

Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/5658/

vvcephei reviewed (approved):
Bummer! Thanks for digging into this, @mjsax .

mjsax commented:
Merged to `trunk` and cherry-picked to `3.5` branch. \cc @mimaison 

Seems some other streams system tests are still flaky (well, I hope it's not a real issue). I hope to find time to dig into it this week.

jolshan commented:
Hey did we mean to remove 3.6 here?

jolshan commented on tests/kafkatest/version.py:
```diff
@@ -249,7 +250,3 @@ def get_version(node=None):
 LATEST_3_5 = V_3_5_0
-
-# 3.6.x versions
-V_3_6_0 = KafkaVersion("3.6.0")
```
Here -- did we mean to remove?

ahuang98 replied:
guessing it was removed because there were no references to it, I'll add it back when I raise a PR for more comprehensive upgrade tests

mjsax replied:
Yes -- 3.6.0 is not released yet. We should only need it after 3.6.0 is out.

If you want to refer to current version it would be `DEV_VERSION` (ie, 3.6.0-SNAPSHOT).

jolshan replied:
Hmm -- if you look at all the snapshot commits, we've added this cutting the release branch for the previous version. (Ie, cutting the branch for the new trunk as per the commit comments) Seems like the top commit I linked should not have been reverted unless all of these were wrong.
https://github.com/apache/kafka/commit/dc1ede8d89d0964783302e0da9ead7fa1d76fbe4
https://github.com/apache/kafka/commit/c1a54671e8fc6c7daec5f5ec3d8c934be96b4989
https://github.com/apache/kafka/commit/6ace67b2de00f4f1665e7c9e3484e01b6d6f9584

jolshan replied:
I think the issue in David's commit was that he didn't set the dev version to match the new version defined?

mjsax replied:
Could be -- I mean, it does not really hurt to add it early (it's just a definition of a var), but it's also useless to have, a nobody should use it, but use `DEV_VERSION` instead.
